ziti/cmd/verify/ops_verify_network.go

```
@@ -0,0 +1,263 @@
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+package verify
+
+import (
+	"fmt"
+	"gopkg.in/yaml.v3"
+	"net"
+	"os"
+	"strings"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"github.com/openziti/ziti/common"
+)
+
+type network struct {
+	controller
+	controllerConfig string
+	routerConfig     string
+	verbose bool
+}
+
+func NewVerifyNetwork() *cobra.Command {
+	n := &network{}
+
+	cmd := &cobra.Command{
+		Use:   "verify-network",
+		Short: "Verifies the overlay is configured correctly",
+		Long:  "A tool to verify network configurations, checking controller and router ports or other common problems",
+		Run: func(cmd *cobra.Command, args []string) {
+			if n.verbose {
+				log.SetLevel(log.DebugLevel)
+			}
+			configureLogFormat()
+			anyFailure := false
+			if n.controllerConfig != "" {
+				log.Infof("Verifying controller config: %s", n.controllerConfig)
+				anyFailure = verifyControllerConfig(n.controllerConfig) || anyFailure
+				fmt.Println()
+			}
+			if n.routerConfig != "" {
+				log.Infof("Verifying router config: %s", n.routerConfig)
+				anyFailure = verifyRouterConfig(n.routerConfig) || anyFailure
+				fmt.Println()
+			}
+			if anyFailure {
+				log.Error("One or more error. Review the output above for errors.")
+			} else {
+				log.Info("All requested checks passed.")
+			}
+		},
+	}
+
+	cmd.Flags().StringVarP(&n.controllerConfig, "controller-config-file", "c", "", "the controller config file verify")
+	cmd.Flags().StringVarP(&n.routerConfig, "router-config-file", "r", "", "the router config file to verify")
+	cmd.Flags().BoolVar(&n.verbose, "verbose", false, "Show additional output.")
+
+	return cmd
+}
+
+type StringMap map[string]interface{}
+
+func (m StringMap) mapFromKey(key string) StringMap {
+	if v, ok := m[key]; ok {
+		return v.(StringMap)
+	}
+	log.Fatalf("map didn't contain key %s", key)
```

It is very odd for a data structure like this to have a fatal error out instead of returning an error and letting the caller decide.

---

changed StringMapList --> stringMapList. unintentionally public